AGENTS.md

```diff
@@ -0,0 +1,799 @@
+
+This is a comprehensive guide for AI coding agents working on the Condo project - an Open Source property management SaaS platform.
+
+## Project Overview
+
+**Condo** is a property management platform that allows users to manage tickets, resident contacts, properties, payment tracking, invoices, and a service marketplace. It offers an extension system for mini-apps and is built as a monorepo with multiple apps and shared packages.
+
+- **Repository:** https://github.com/open-condo-software/condo
+- **License:** MIT
+- **Architecture:** Monorepo with Yarn workspaces
+- **Primary Stack:** KeystoneJS, NextJS, ApolloGraphQL, PostgreSQL, Redis
+
+## Monorepo Structure
+
+### Apps (`/apps/*`)
+Independent applications that cannot use code from each other. Each app is a standalone service:
+- `condo` - Main Keystone + Next web application (property management)
+- `pos-integration` - POS integration service
+- `billing-connector` - Billing integration
+- `callcenter` - Call center service
+- `resident-app` - Resident mobile/web app
+- `dev-portal-web` - Developer portal
+- And 20+ other specialized apps
+
+### Packages (`/packages/*`)
+Internal libraries shared across apps:
+- `@open-condo/ui` - **UI Kit** (recommended for all GUI elements)
+- `@open-condo/icons` - Icon library
+- `@open-condo/*` packages - Core utilities and shared functionality
+- `webhooks` - Webhook functionality
+- `apollo` - Apollo client utilities
+- `billing` - Billing utilities
+- And more specialized packages
+
+## Environment Requirements
+
+### Required Software
+- **Node.js:** 22.x (LTS) - Use nvm for local development
+- **Yarn:** 3.2.2+ (Berry)
+- **Python:** 3.x (for database migrations)
+- **PostgreSQL:** 16.4+
+- **Redis:** 6.2+
+- **Docker & Docker Compose** (optional, for databases)
+
+### Python Packages
+```bash
+pip install Django psycopg2-binary
+```
+
+## Setup Commands
+
+### Initial Setup
+```bash
+# 1. Start databases (optional, using Docker)
+docker compose up -d postgresdb redis
+
+# 2. Install Node.js dependencies
+yarn install
+
+# 3. Build @open-condo packages (required before running apps)
+yarn workspace @app/condo build:deps
+
+# 4. Prepare local environment (creates .env, databases, test users)
+node bin/prepare -f condo
+```
+
+### Development Commands
+```bash
+# Start main condo app in dev mode
+yarn workspace @app/condo dev
+
+# Start worker (handles async tasks)
+yarn workspace @app/condo worker
+
+# Run tests
+yarn workspace @app/condo test
+yarn workspace @app/condo test User.test.js  # Specific test file
+
+# Build for production
+yarn workspace @app/condo build
+yarn workspace @app/condo start
+```
+
+### Database Migrations
+```bash
+# Run migrations
+yarn workspace @app/condo migrate
+
+# Create new migrations after schema changes
+yarn workspace @app/condo makemigrations
+
+# Revert last migration
+yarn workspace @app/condo migrate:down
+
+# Unlock migrations table
+yarn workspace @app/condo migrate:unlock
+```
+
+### Build Commands
```

Comment:
Add command to build dependencies `build:deps`

Reply:
done